packages/paste-website/src/pages/components/truncate/index.mdx

```diff
@@ -0,0 +1,193 @@
+        depth
+        value
+      }
+    }
+  }
+`;
+
+<ComponentHeader
+  name={props.pageContext.frontmatter.title}
+  categoryRoute={SidebarCategoryRoutes.COMPONENTS}
+  githubUrl="https://github.com/twilio-labs/paste/tree/main/packages/paste-core/components/truncate"
+  storybookUrl="https://twilio-labs.github.io/paste/?path=/story/components-truncate--default"
+  data={props.data.allPasteComponent.edges}
+/>
+
+---
+
+<contentwrapper>
+
+<PageAside data={props.data.mdx} />
+
+<content>
+
+## Guidelines
+
+### About Truncate
+
+Truncation, or shortening, is typically used for text or links that exceed the size of their container.
+The Truncate component does all of the heaving lifting for you. You just need to wrap the long
+text with the component, and an element with a set width. This will add cut off the text based on
+the parent containers width, and append an ellipsis `...` at the end.
+
+Good use cases for truncation include:
+
+- Breadcrumbs
+- Long URL links
+- Long item names
+- Long account numbers
+
+<Callout variant="warning">
+  <CalloutText>
+    Truncation should <strong>NOT</strong> be used on page headers, titles, labels, error messages, validation messages,
+    or notifications.
+  </CalloutText>
+</Callout>
+
+#### Accessibility
+
+By default, Truncate has no accessibility concerns. While using Truncate, it is up to you to manage the resulting
+accessibility implications. You may want to consider how the Truncate affects readability due to shortening of
+information on your page.
+
+## Examples
+
+### Basic Truncate
+
+<LivePreview
+  scope={{
+    Truncate,
+    Box,
+    Text,
+  }}
+>
+  {`<Box maxWidth="size30">
+  <Text as="p">
+    <Truncate>Black Lives Matter. We stand with the Black community.</Truncate>
+  </Text>
+</Box>`}
+</LivePreview>
+
+### Truncate URL
+
+<LivePreview
+  scope={{
+    Truncate,
+    Anchor,
+    Box,
+  }}
+>
+  {`<Box maxWidth="size30">
+  <Anchor href="/getting-started/about-paste">
+    <Truncate>https://paste.twilio.design/getting-started/about-paste</Truncate>
+  </Anchor>
+</Box>`}
+</LivePreview>
+
+<Callout variant="warning">
+  <CalloutTitle>🚨Don't forget the container element 🚨</CalloutTitle>
+  <CalloutText>
+    Truncate must be used in a container with a set width. Without this container Truncate will have no effect since it
+    spans the full width of a container.
+  </CalloutText>
+</Callout>
+
+<DoDont>
+  <Do title="Do" body="Use to shorten text that doesn't fit in its parent container">
+    <Box maxWidth="size20" padding="space60">
+      <Text as="p">
+        <Truncate>Some very long text to truncate</Truncate>
+      </Text>
+    </Box>
+  </Do>
+  <Dont
+    title="Don't"
+    body="Use to shorten page headings, titles, labels, error messages, validation messages,
+    or notifications"
+  >
+    <Box maxWidth="size20" padding="space60">
+      <Heading as="h3" variant="heading30">
+        <Truncate>A very important long page heading</Truncate>
+      </Heading>
```

Comment:
Might be more impactful to show this as an Alert, so it's more immediately obvious why truncate shouldn't be used in these cases.